### PR TITLE
signedexchange: Add flags 'requestHeader' and 'responseHeader'

### PR DESCRIPTION
**Pull request by hajimehoshi**
#124 

This is WIP: I have no idea how to treat the passed request headers.

### CODE REVIEW

**Comment by jyasskin** [comment]
Right now, the spec says not to sign request headers, which would imply that the tool shouldn't actually take a `--requestHeader` argument. If we want to sign request headers (which is plausible, especially inside `application/http-exchange+cbor` objects instead of plain HTTP responses), I think the natural thing is to include them in the map created by `encodeRequest()`.

**Comment by hajimehoshi** [comment]
> I think the natural thing is to include them in the map created by encodeRequest().

Thank you, done.

File: go/signedexchange/signedexchange.go [hunk 1]
```diff
@@ -97,6 +99,13 @@ func (e *Exchange) encodeRequest(enc *cbor.Encoder) error {
 			valueE.EncodeByteString([]byte(e.requestUri.String()))
 		}),
 	}
+	for name, value := range e.requestHeaders {
```
Comment (nyaxt) [inline go/signedexchange/signedexchange.go]: encodeRequest from L172 should include requestHeaders, while L149 should not

Comment (hajimehoshi) [reply go/signedexchange/signedexchange.go]: Done.

**Pull request merged**